### ISSUE TITLE
Can't reach the 8123 port

### ISSUE THREAD

**Issue by arunoruto** [bigmoby/hassio-repository-addon#3]
Nice add on and really easy to setup. I wanted to use to connect to my VPS and then access my home assistant instance over the server. I am doing it currently using an SSH Tunnel add-on, but wanted to switch to wireguard since I can deploy one container on my server for multiple services.

Well, in trying to achieve this I wanted to simply try first if I can ping my home assistant through the tunnel and yeah, it works. But now things get weird. I tried `curl` on the `8123` port and it does nothing. Like it just waits and that is it. Using curl on port `4357` gives me an output tho! I am also able to curl the `8123` port inside the add-on container, but not on the container of my server. Am I missing something inside home assistant?

*EDIT* added some highlighting.

**Comment by bigmoby** [comment]
Hi @SoulEater45 , glad to hear that the Wireguard Client add-on is useful for you 👍
I'm pretty sure that some redirect rule is missing or wrong, please try to follow this guide I wrote https://github.com/bigmoby/homeassistant_nat_network and the second part https://github.com/bigmoby/homeassistant_nat_network/blob/main/README_IT_part_2.md
I'm sorry it's in Italian only but, trust me, it's very didascalic and easy to translate (Google translate for example).
Please contribute to this project of it's a good tool for you.
All the best,
Fabio

**Comment by arunoruto** [comment]
Hello Fabio, thanks for the quick reply. It is nice to see when a dev is fast in reaching out ;)
I have read the linked articles and it was quite easy with the translation. My setup differs a bit, since I am using your add-on (which is a docker container) to connect to a wireguard instance also hosted on my server using the [linuxserver container](https://hub.docker.com/r/linuxserver/wireguard). I have tried all kinds of configurations and routing and bla bla, but nothing has worked on the port `8123`. On contrary, `4357` can be `curl`ed without any problems. And what I mean by `curl`ed, is that I am using it inside the containers, not outside! Is there maybe some kind of home assistant setting I am missing? Like the `trusted_reverse_proxy` in `http` or similar? Like for the sake of god, I am not able to get anything on port `8123` on the server :(

**Comment by arunoruto** [comment]
Okay, never mind, I kinda got it working. I went a step back and looked with `nmap` inside the container if the port can be seen. All ports were available, but `8123` and `9000` were filtered. That got me thinking. In previous attempts to make the thing run, I also used the port `9000` for some testing, so maybe I broke something... After a restart of the host device (my Pi4) and a barebones connection setup on the "client" side, I got it working! Seems like I messed something up in the network (dunno how), but now I got my port `8123` in the network and I can also mapped to the container without any issues :)
